src/utils/domain-utils.js

```diff
@@ -0,0 +1,36 @@
+import { ALL_FIELD_TYPES } from "constants";
+import { unique, notNullorUndefined } from "utils/data-utils";
+
+export function mergeDomain(oldDomain, newDomain, fieldType) {
+  switch (fieldType) {
+    case ALL_FIELD_TYPES.real:
+    case ALL_FIELD_TYPES.integer:
+    case ALL_FIELD_TYPES.timestamp:
+      return mergeNumericFieldDomain(oldDomain, newDomain);
+    case ALL_FIELD_TYPES.boolean:
```

Comment:
boolean field domain is always true or false, so you can just return the `oldDomain` to avoid triggering `updateTriggers`